### DISCUSSION TITLE
Why isn't openzeppelin and oz upgradable added to the boilerplate?

### DISCUSSION THREAD

**Discussion by davidkuchar** [PaulRBerg/hardhat-template#34] [Q&A]
Just looking for your opinion on that - really nice job on this!

**Comment by PaulRBerg** [comment]
Thanks, glad you like the template!

I personally think that [contract upgradeability is a bug, not a feature](https://twitter.com/PaulRBerg/status/1325432145947537415). I stand with Steve Marx's [take](https://medium.com/consensys-diligence/upgradeability-is-a-bug-dba0203152ce) on this.

As with regards to [openzeppelin-contracts](https://github.com/OpenZeppelin/openzeppelin-contracts), I shared my thoughts [here](https://github.com/paulrberg/solidity-template/issues/29#issuecomment-806466494). However, I am becoming more open to it.

I may add OZ later down the line - anyone can signal support by upvoting this comment :)

**Reply by davidkuchar** [reply]
thanks for the details on this - I'll take a look!
